File: Common/Models/UserChartModel.cs
namespace Common.Models
{
    public class UserChartModel
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }

        public string MqttTopic { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; }

        public IEnumerable<SensorWithValuesModel> Sensors { get; set; } = new List<SensorWithValuesModel>();

    }
}

File: SPA/ClientApp/src/app/models/UserChartModel.ts
import { SensorWithValuesModel } from "./SensorWithValuesModel";

export class UserChartModel {
    id!: number;
    name!: string;
    description: string | null = null;
    mqttTopic!: string;
    createdAt: string = '';
    sensors: SensorWithValuesModel[] = [];
}